Content.Server/MachineLinking/System/DeadMansSwitchSystem.cs

```diff
@@ -0,0 +1,127 @@
+﻿using Content.Server.DoAfter;
+using Content.Server.MachineLinking.Components;
+using Content.Server.Popups;
+using Content.Shared.Interaction.Events;
+using Content.Shared.MobState;
+using Content.Shared.Verbs;
+using Robust.Shared.Player;
+
+namespace Content.Server.MachineLinking.System;
+
+/// <summary>
+/// Entity system for dead man's switches.
+/// </summary>
+public sealed class DeadMansSwitchSystem : EntitySystem
+{
+    [Dependency] private readonly SignalLinkerSystem _signalSystem = default!;
+    [Dependency] private readonly PopupSystem _popupSystem = default!;
+    [Dependency] private readonly DoAfterSystem _doAfterSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<DeadMansSwitchComponent, ComponentInit>(OnInit);
+        SubscribeLocalEvent<DeadMansSwitchComponent, DroppedEvent>(OnDropped);
+        SubscribeLocalEvent<DeadMansSwitchComponent, UseInHandEvent>(OnUsedInHand);
+        SubscribeLocalEvent<DeadMansSwitchComponent, GetVerbsEvent<AlternativeVerb>>(OnGetVerbs);
+        SubscribeLocalEvent<DeadMansSwitchComponent, DeadMansSwitchDisarmCompleteEvent>(OnSuccessfulDisarm);
+
+        SubscribeLocalEvent<DeadMansSwitchHolderComponent, MobStateChangedEvent>(OnMobStateChanged);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+        foreach (var holder in EntityQuery<DeadMansSwitchHolderComponent>())
+        {
+            if (holder.Switches.Count == 0)
+            {
+                RemComp<DeadMansSwitchHolderComponent>(holder.Owner);
+            }
+        }
+    }
+
+    private void OnInit(EntityUid uid, DeadMansSwitchComponent component, ComponentInit args)
+    {
+        _signalSystem.EnsureTransmitterPorts(uid, component.Port);
+    }
+
+    private void OnDropped(EntityUid uid, DeadMansSwitchComponent component, DroppedEvent args)
```

Comment:
Be careful with this because this event raises when you move items to backpack / pockets / etc.

Reply:
I mean, you're taking it out of your hand, so, working as intended I guess.